spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -32,21 +33,37 @@
  * @since 1.0
  *
  */
-public class UnZipResultSplitter {
-
-	public List<Message<Object>> splitUnzippedMap(Map<String, Object> unzippedEntries) {
+public final class UnZipResultSplitter {
 
-		final List<Message<Object>> messages = new ArrayList<Message<Object>>(unzippedEntries.size());
+	public List<Message<Object>> splitUnzippedMap(Message<Map<String, Object>> message) {
+		return createMessages(message.getPayload(), message.getHeaders());
+	}
 
+	private List<Message<Object>> createMessages(Map<String, Object> unzippedEntries, MessageHeaders headers) {
+		List<Message<Object>> messages = createList(unzippedEntries.size());
 		for (Map.Entry<String, Object> entry : unzippedEntries.entrySet()) {
-			final String path = FilenameUtils.getPath(entry.getKey());
-			final String filename = FilenameUtils.getName(entry.getKey());
-			final Message<Object> splitMessage = MessageBuilder.withPayload(entry.getValue())
-					.setHeader(FileHeaders.FILENAME, filename)
-					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path).build();
-			messages.add(splitMessage);
+			messages.add(createMessage(entry, headers));
 		}
 		return messages;
 	}
 
+	private static List<Message<Object>> createList(int initialCapacity) {
+		return new ArrayList<Message<Object>>(initialCapacity);
+	}
+
+	private static Message<Object> createMessage(Map.Entry<String, Object> entry, Map<String, ?> headers) {
+		return MessageBuilder.withPayload(entry.getValue())
+                .setHeader(FileHeaders.FILENAME, getName(entry))
+                .setHeader(ZipHeaders.ZIP_ENTRY_PATH, getPath(entry))
+				.copyHeaders(headers)
+                .build();
+	}
+
+	private static String getName(Map.Entry<String, Object> entry) {
+		return FilenameUtils.getName(entry.getKey());
+	}
+
+	private static String getPath(Map.Entry<String, Object> entry) {
+		return FilenameUtils.getPath(entry.getKey());
```

Comment:
Again: any utility method introduction should bring its general purpose. this two methods just do a single particular operators. We should think about optimization ASAP. IMO double getKey() invocation is overhead. Plus I don't see so much gain for this methods. They don't provide comprehensive logic to be severed from the main method. From other hand following with your logic , it doesn't make code understable, at least for me... So, since I have a responsibility to review and merge your code to the public community wide repository, you should convince me in these methods. Sorry that I don't think like you, but it really looks like we are fruits of different fields 😀